### PR TITLE
BUG/STY: fix GOLD updates

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Several bugfixes were pushed to develop for GOLD data while #161 was under review.  Additionally, some data products were changed to use the cdasws functions for download.  2 of the 3 new products in #161 could use this, but the third was not in the cdasws database. 

Updates #161 with latest bugfixes, style.  Note that full functionality requires pysat 3.1.0.  Basic functionality (loading single days, not multiple days) is maintained for earlier pysat versions, but will likely be re-addressed after pysat 3.1.0 is released.

## Type of change

- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Load new gold data.

## Test Configuration
* Operating system: Monterrey
* Version number: python 3.9.7
* Any details about your local setup that are relevant

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes
- [x] Update zenodo.json file for new code contributors

### CODE REVIEW

File: pysatNASA/instruments/ses14_gold.py [hunk 1]
```diff
@@ -42,6 +43,7 @@
 import numpy as np
 import xarray as xr
```
Comment (jklenzing) [inline pysatNASA/instruments/ses14_gold.py:44]:
```suggestion
```

File: pysatNASA/instruments/ses14_gold.py [hunk 2]
```diff
@@ -165,14 +249,23 @@ def load(fnames, tag='', inst_id=''):
     elif tag == 'o2den':
         epoch_name = 'nevents'
 
-    data, meta = load_netcdf(fnames, pandas_format=pandas_format,
-                             epoch_name=epoch_name, labels=labels,
-                             meta_translation=meta_translation,
-                             combine_by_coords=False,
-                             drop_meta_labels='FILLVAL')
+    try:
```
Comment (jklenzing) [inline pysatNASA/instruments/ses14_gold.py:252]:
```suggestion
    # TODO(#165): remove try/except notation once pysat 3.1.0 is released
    try:
```

File: pysatNASA/instruments/ses14_gold.py [hunk 3]
```diff
@@ -87,13 +89,95 @@
...
+        (default=None)
+    stop : dt.datetime or NoneType
+        Ending time for the file list.  A None value will stop with the last
+        file found.
+        (default=None)
+    supported_tags : dict
+        dict of dicts. Keys are supported tag names for download. Value is
+        a dict with 'remote_dir', 'fname'. Inteded to be
+        pre-set with functools.partial then assigned to new instrument code.
+        (default=None)
```
Comment (jklenzing) [inline pysatNASA/instruments/ses14_gold.py:158]:
```suggestion
```

**Review by aburrell — COMMENTED** [review]
Are the windows tests expected to fail?

**Comment by jklenzing** [comment]
> Are the windows tests expected to fail?

Windows 10 will fail until pysat 3.1.0 is released.

File: pysatNASA/instruments/ses14_gold.py [hunk 4]
```diff
@@ -87,13 +89,90 @@
...
+    ----------
+    date_array : array-like
+        Array of datetimes to download data for. Provided by pysat.
+    tag : str
+        Data product tag (default='')
+    inst_id : str
+        Instrument ID (default='')
+    data_path : str or NoneType
+        Path to data directory.  If None is specified, the value previously
+        set in Instrument.files.data_path is used.  (default=None)
```
[the remark this answers is not in the document]
Comment (jklenzing) [reply pysatNASA/instruments/ses14_gold.py:120]: `data_path=None` interfaces with pysat directly.  If this has been updated, we'll need to fix this in the methods files as well.  Probably worth an issue.

[the remark this answers is not in the document]
Comment (aburrell) [reply pysatNASA/instruments/ses14_gold.py:120]: Yeah, pysatSpaceWeather has `data_path` as an arg, and I know we had a project to clear this up at some point.  An issue here is probably worthwhile.

[the remark this answers is not in the document]
Comment (jklenzing) [reply pysatNASA/instruments/ses14_gold.py:120]: Opened #166 to capture this.

**Review by aburrell — APPROVED** [review]

**Pull request merged**